Explorer/Assets/DCL/UI/MainUIContainer/MainUIController.cs

```diff
@@ -0,0 +1,156 @@
+
+            if (!showingSidebar && !waitingToShowSidebar)
+            {
+                waitingToShowSidebar = true;
+                showSidebarCancellationTokenSource = showSidebarCancellationTokenSource.SafeRestart();
+                WaitAndShow(showSidebarCancellationTokenSource.Token).Forget();
+            }
+
+            if (waitingToHideSidebar) { hideSidebarCancellationTokenSource.Cancel(); }
+        }
+
+        private void OnPointerExit()
+        {
+            if (autoHideSidebar || sidebarBlockStatus) return;
+
+            if (waitingToShowSidebar || showingSidebar) { showSidebarCancellationTokenSource.Cancel(); }
+
+            if (hideSidebarCancellationTokenSource.IsCancellationRequested) { waitingToHideSidebar = false; }
+
+            if (!waitingToHideSidebar && showingSidebar)
+            {
+                waitingToHideSidebar = true;
+                hideSidebarCancellationTokenSource = hideSidebarCancellationTokenSource.SafeRestart();
+                WaitAndHide(hideSidebarCancellationTokenSource.Token).Forget();
+            }
+        }
+
+        private async UniTask WaitAndHide(CancellationToken ct)
+        {
+            await UniTask.Delay(TimeSpan.FromSeconds(HIDE_SIDEBAR_WAIT_TIME), cancellationToken: ct);
+            waitingToHideSidebar = false;
+            if (ct.IsCancellationRequested) return;
+
+            await AnimateWidthAsync(HIDE_SIDEBAR_LAYOUT_WIDTH, ct);
+
+            if (ct.IsCancellationRequested) return;
+            showingSidebar = false;
+        }
+
+        private async UniTask WaitAndShow(CancellationToken ct)
+        {
+            await UniTask.Delay(TimeSpan.FromSeconds(SHOW_SIDEBAR_WAIT_TIME), cancellationToken: ct);
+            waitingToShowSidebar = false;
+            if (ct.IsCancellationRequested) return;
+
+            await AnimateWidthAsync(SHOW_SIDEBAR_LAYOUT_WIDTH, ct);
+
+            if (ct.IsCancellationRequested) return;
+            showingSidebar = true;
+        }
+
+        private async UniTask AnimateWidthAsync(float width, CancellationToken ct)
+        {
+            float startWidth = viewInstance.sidebarLayoutElement.preferredWidth;
+            float endWidth = width;
+            var elapsedTime = 0f;
+
+            while (elapsedTime < SIDEBAR_ANIMATION_TIME)
+            {
+                if (ct.IsCancellationRequested)
+                {
+                    viewInstance.sidebarLayoutElement.preferredWidth = startWidth;
+                    return;
+                }
+
+                elapsedTime += Time.deltaTime;
+                float t = Mathf.Clamp01(elapsedTime / SIDEBAR_ANIMATION_TIME);
+                viewInstance.sidebarLayoutElement.preferredWidth = Mathf.Lerp(startWidth, endWidth, t);
+                await UniTask.Yield();
+            }
+
+            viewInstance.sidebarLayoutElement.preferredWidth = endWidth;
+        }
```

Comment:
Same as before... Maybe it worth it to think on make this animation through DOTween.